### PR TITLE
Task manager -Brian Keane

### PR DESCRIPTION
**Pull request by briankeane**

### CODE REVIEW

File: lib/task-manager/project.rb [hunk 1]
```diff
@@ -1,3 +1,54 @@
...
+  end
+
+  def completed_tasks  #returns completed_tasks in the proper order
+    completed_tasks = @tasks.select { |k,v| v.finished? == true }.values
+    return completed_tasks.sort { |a,b| a.creation_date <=> b.creation_date }
+  end
+
+  def ongoing_tasks  #returns ongoing_tasks in the proper order
+    ongoing_tasks = @tasks.select { |k,v| v.finished? == false }.values
+    ongoing_tasks.sort! { |a,b| (a.priority <=> b.priority) == 0 ? (a.creation_date <=> b.creation_date) : (b.priority <=> a.priority) }
```
Comment (gilbert) [inline lib/task-manager/project.rb:49]: You can simplify this by checking for equality:

ongoing_tasks.sort! { |a,b| (a.priority == b.priority) ? (a.creation_date <=> b.creation_date) : (b.priority <=> a.priority) }

Comment (briankeane) [inline lib/task-manager/project.rb:49]: AAHH!! Got it :)

On Wed, Mar 19, 2014 at 3:10 PM, Gilbert notifications@github.com wrote:

> In lib/task-manager/project.rb:
> 
> > -    # add the task to the @tasks hash
> > -    @tasks[task_object.task_id] = task_object
> > -  end
> >   +
> > -  def mark_as_finished(task_id)
> > -    @tasks[task_id].finished = true
> > -  end
> >   +
> > -  def completed_tasks  #returns completed_tasks in the proper order
> > -    completed_tasks = @tasks.select { |k,v| v.finished? == true }.values
> > -    return completed_tasks.sort { |a,b| a.creation_date <=> b.creation_date }
> > -  end
> >   +
> > -  def ongoing_tasks  #returns ongoing_tasks in the proper order
> > -    ongoing_tasks = @tasks.select { |k,v| v.finished? == false }.values
> > -    ongoing_tasks.sort! { |a,b| (a.priority <=> b.priority) == 0 ? (a.creation_date <=> b.creation_date) : (b.priority <=> a.priority) }
> 
> You can simplify this by checking for equality:
> 
> ongoing_tasks.sort! { |a,b| (a.priority == b.priority) ? (a.creation_date
> <=> b.creation_date) : (b.priority <=> a.priority) }
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/makersquare/ruby-course/pull/101/files#r10769010
> .

## 

www.briankeane.net
www.myspace.com/briankeane

File: lib/task-manager/pm.rb [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+      puts"\tshow PID - Show remaining tasks for project with id=PID\n"
+      puts"\thistory PID - Show completed tasks for project with id=PID\n"
+      puts"\tmark TID - Mark task with id=TID as complete\n"
+      puts"\tload me up -- Make a bunch of fake projects so you look busy.\n"
+      puts"\tquit -- quit this program\n\n"
+      print"> "
+      choice = gets.chomp.downcase
+
+      # take care of those pesky 2 word ones.
+      if choice.split(' ').size > 1
```
Comment (provostprovost) [inline lib/task-manager/pm.rb:34]: We didn't create the control/choice array outside of our individual functions, way to DRY

File: lib/task-manager/pm.rb [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+        when "add"
+          if choice_array[1] == "project"
+            self.add_project
+          elsif choice_array[1] == "task"
+            self.add_task
+          else
+            puts "Add what, dude?"
+          end
+        else
+          if (  (choice_array.include?("shit"))  ||
```
Comment (provostprovost) [inline lib/task-manager/pm.rb:54]: Forgot the c word

File: lib/task-manager/pm.rb [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+
+
+  def history(project_id)
+    project = TM::Project.all_projects[project_id]
+    tasks_array = project.completed_tasks
+    puts "\n\nProject: #{project.name}\n"
+    puts "ID\tDescription\t\t\tPriority\n"
+    puts "------------------------------------------------"
+    tasks_array.each { | x | print("#{x.task_id}" + (' ' * (8 - x.task_id.to_s.length)) + # padding
```
Comment (provostprovost) [inline lib/task-manager/pm.rb:113]: Cool formatting bro

File: lib/task-manager/project.rb [hunk 5]
```diff
@@ -1,3 +1,54 @@
-
 class TM::Project
+  attr_reader :name, :id, :tasks
+
+  @@current_id = 1
+  @@all_projects = {}
```
Comment (provostprovost) [inline lib/task-manager/project.rb:5]: I think storing the array/hash of all projects makes more sense in its own class.

File: lib/task-manager/task.rb [hunk 6]
```diff
@@ -1,3 +1,35 @@
-
 class TM::Task
+  attr_accessor :project_id, :description, :priority, :finished
+  attr_reader :task_id, :creation_date
+  @@current_id = 1
+  @@all_tasks = {}
```
Comment (provostprovost) [inline lib/task-manager/task.rb:5]: Same as my previous comment. In my mind, A Projects List should store all Projects, and each Project should store all Tasks associated with the project. I don't even know if that is correct, it is just how we have been thinking about it

File: lib/task-manager/project.rb (hunk 1 above)
Comment (provostprovost) [inline lib/task-manager/project.rb:49]: We used sort_by for this. Mostly because I am not sure what <=> does or means

File: spec/project_spec.rb [hunk 7]
```diff
@@ -4,4 +4,70 @@
   it "exists" do
     expect(TM::Project).to be_a(Class)
   end
+
+  before do
+    TM::Project.current_id = 1
+    @kill_bob = TM::Project.new("Kill Bob")
```
Comment (provostprovost) [inline spec/project_spec.rb:10]: Don't do it, think of your family

**Comment by provostprovost** [comment]
Looks pretty good to me.

**Comment by gilbert** [comment]
Great job, Brian.

**Pull request closed without merging**